### PR TITLE
Support "iterations" option

### PR DESCRIPTION
**Pull request by bengl**
### What does this PR do?
- support 'iterations' option

In JSON/YAML config, this must be set to an integer > 0 or not set at all, which
the default of 1 is used. This instructs sirun to run the test that many times.

This adds an `iterations` field to the output JSON when more than one iteration
is run. Iteration-specific metrics are included here. If only one iteration is
run, the results will instead be at the top level.

### Motivation

Many tests would benefit from collecting more data without having to run the
setup task more than once.

### Checklist

[x] I have added tests for the code I am submitting
[x] My unit tests cover both failure and success scenarios
[] If applicable, I have discussed my architecture

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -185,8 +176,39 @@ async fn main() -> Result<()> {
...
+    statsd_started.wait().await; // waits for socket to be listening
 
     let mut metrics: HashMap<String, MetricValue> = HashMap::new();
+    if config.iterations == 1 {
+        run_iteration(&config, &mut metrics, statsd_buf.clone()).await?;
+    } else {
+        let mut iterations = Vec::new();
+        for _ in 0..config.iterations {
+            let mut iteration_metrics = HashMap::new();
+            run_iteration(&config, &mut iteration_metrics, statsd_buf.clone()).await?;
```
Comment (brettlangdon) [inline src/main.rs:207]: I am likely over-engineering, but should we inject some SIRUN env variables into the command?

`SIRUN_ITERATION=5` ?

(non-blocking comment, random thought)

**Review by brettlangdon — APPROVED** [review]

File: src/main.rs (hunk 1 above)
Comment (bengl) [reply src/main.rs:207]: Is there a use case where you'd want to do something different for each iteration? The results are retrieved serially, so if it's only to include in output, you already have the ordering of the array.

Comment (brettlangdon) [reply src/main.rs:207]: yeah, I suppose I saw `echo "this is something"` and thought... hey, `echo "iteration: ${SIRUN_ITTERATION}"` but I suppose doesn't really add much value other than that 😆

Comment (bengl) [reply src/main.rs:207]: Yeah I think if you want to switch based on it, you probably actually want a variant.

Running all variants seems like a desirable feature, but out of scope for this PR.

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**